front/app/utils/dateUtils.test.ts

```diff
@@ -0,0 +1,187 @@
+import {
+  pastPresentOrFuture,
+  getIsoDateUtc,
+  getIsoDate,
+  getIsoDateForToday,
+} from './dateUtils';
+import moment from 'moment';
+import 'moment-timezone';
+
+// test date is 1AM June 15 2020 UTC time (Z)
+const testDateStr = '2020-06-15T01:00:00Z';
+
+// the local time in Tokyo should be June 14th 2022, 9pm,
+// since Tokyo is -0400 hour offset from UTC time
+describe('in America/Santiago time zone', () => {
+  beforeAll(() => {
+    jest.useFakeTimers('modern');
+    jest.setSystemTime(new Date(testDateStr));
+    moment.tz.setDefault('America/Santiago');
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  describe('general time safety checks', () => {
+    it('gets the day and time correctly for the America/Santiago timezone', () => {
+      expect(
+        new Date().toLocaleString('en-US', { timeZone: 'America/Santiago' })
+      ).toBe('6/14/2020, 9:00:00 PM');
+    });
+  });
+
+  describe('getIsoDateUtc', () => {
+    it('returns the correct date formatted, ignoring timezone', () => {
+      expect(getIsoDateUtc('2020-06-15')).toEqual('2020-06-15');
+    });
+  });
+
+  describe('getIsoDate', () => {
+    it('returns the correct date formatted, taking into account user timezone', () => {
+      expect(getIsoDate('2020-06-15')).toEqual('2020-06-14');
+    });
+  });
+
+  describe('getIsoDateForToday', () => {
+    it('returns the correct date formatted', () => {
+      expect(getIsoDateForToday()).toEqual('2020-06-14');
+    });
+  });
+
+  describe('pastPresentOrFuture', () => {
+    describe('with a single input date', () => {
+      it('should accurately report a past date', () => {
+        expect(pastPresentOrFuture('2020-06-13')).toEqual('past');
+      });
+
+      it('should accurately report a present date', () => {
+        expect(pastPresentOrFuture('2020-06-14')).toEqual('present');
```

Comment:
How is this 'present'? The date you used is 15/6/2020, right?

Reply:
For the user in Santiago timezone, 1am UTC time is 9pm on the 14th (-4 hours offset) so 06-14 should show up as present. Same as below, the 15th should be in 3 hours so that will be "future."

Reply:
Ok, I mixed things up here :)